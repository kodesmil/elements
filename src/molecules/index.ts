export * from './dashboard/ks-dashboard-header'
export * from './dashboard/ks-dashboard-table'
export * from './dashboard/ks-dropdown-menu'

export * from './forms/ks-form-actions'
export * from './forms/ks-form-header'
export * from './forms/ks-form-layout'
export * from './forms/ks-form-section'
export * from './forms/ks-form-subheader'
export * from './forms/ks-form-subsection'

export * from './forms/fields/ks-date-field'
export * from './forms/fields/ks-date-time-field'
export * from './forms/fields/ks-name-field'
export * from './forms/fields/ks-picture-field'
export * from './forms/fields/ks-input-field'
export * from './forms/fields/ks-text-field'
export * from './forms/fields/ks-select-field'
export * from './forms/fields/ks-single-choice-field'
export * from './forms/fields/ks-switch-field'
export * from './forms/fields/ks-multi-choice-field'
export * from './forms/fields/ks-multi-selector-field'
export * from './forms/fields/ks-text-area-field'
export * from './forms/fields/ks-text-editor-field'

export * from './pages/ks-page-header'
export * from './pages/ks-page-header-actions'
export * from './pages/ks-page-layout'
export * from './pages/ks-page-list'
export * from './pages/ks-page-section'
export * from './pages/ks-page-section-header'

export * from './text-editor/text-editor'
export * from './text-editor/extensions'

export * from './ks-breadcrumb'
export * from './ks-card-header-layout'
export * from './ks-copy-button'
export * from './ks-header'
export * from './ks-icon'
export * from './ks-image'
export * from './ks-layout'
export * from './ks-link-icon'
export * from './ks-search'
export * from './ks-sheet'
export * from './ks-sidebar'
export * from './ks-tooltip'
export * from './ks-image'
