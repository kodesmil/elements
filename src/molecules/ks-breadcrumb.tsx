import {
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  BreadcrumbList,
  BreadcrumbPage,
  BreadcrumbSeparator,
} from '@/atoms/shacdn/breadcrumb'
import Link from 'next/link'
import React from 'react'

export function KsBreadcrumb(props: {
  breadcrumbs: {
    label: string
    href: string
  }[]
}) {
  return (
    <Breadcrumb className="hidden md:flex">
      <BreadcrumbList>
        {props.breadcrumbs.map((link, index) => (
          <React.Fragment key={index}>
            <BreadcrumbItem>
              <BreadcrumbLink asChild={true}>
                {index !== props.breadcrumbs.length - 1 ? (
                  <Link href={link.href}>{link.label}</Link>
                ) : (
                  <BreadcrumbPage>{link.label}</BreadcrumbPage>
                )}
              </BreadcrumbLink>
            </BreadcrumbItem>
            {index !== props.breadcrumbs.length - 1 && <BreadcrumbSeparator />}
          </React.Fragment>
        ))}
      </BreadcrumbList>
    </Breadcrumb>
  )
}
