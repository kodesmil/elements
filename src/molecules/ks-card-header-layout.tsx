import type { ReactNode } from 'react'

export function KsCardHeaderLayout(props: {
  children: ReactNode
  actions: ReactNode
}) {
  return (
    <div className={'flex flex-row justify-between'}>
      <div className={'flex flex-col gap-3'}>{props.children}</div>
      <div className={'flex flex-row'}>{props.actions}</div>
    </div>
  )
}
