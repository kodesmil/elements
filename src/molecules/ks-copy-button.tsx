import { Button } from '@kodesmil/elements/atoms'
import { type FC, useState } from 'react'

interface KsCopyButtonProps {
  textToCopy: string
  copyButtonName?: string
  copiedButtonName?: string
}

export const KsCopyButton: FC<KsCopyButtonProps> = ({
  textToCopy,
  copyButtonName = 'Copy',
  copiedButtonName = 'Copied!',
}) => {
  const [isCopied, setIsCopied] = useState(false)

  const handleCopy = async () => {
    try {
      await navigator.clipboard.writeText(textToCopy)
      setIsCopied(true)
      setTimeout(() => setIsCopied(false), 3000)
    } catch (_error) {}
  }

  return (
    <Button variant={'secondary'} onClick={handleCopy}>
      {isCopied ? copiedButtonName : copyButtonName}
    </Button>
  )
}
