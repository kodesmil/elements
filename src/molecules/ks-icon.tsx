import type { LucideProps } from 'lucide-react'
import dynamicIconImports from 'lucide-react/dynamicIconImports'
import dynamic from 'next/dynamic'

export type KsIconNameType = keyof typeof dynamicIconImports

interface KsIconProps extends LucideProps {
  name: KsIconNameType
}

export const KsIcon = ({ name, ...props }: KsIconProps) => {
  const LucideIcon = dynamic(dynamicIconImports[name])

  return <LucideIcon {...props} />
}
