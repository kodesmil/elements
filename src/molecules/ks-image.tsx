import KsIcon, { type KsIconNameType } from '@/molecules/ks-icon'
import Image, { type ImageProps } from 'next/image'

export const KsImage = (
  props: {
    src?: string
    fallbackIconName?: KsIconNameType
  } & ImageProps
) => {
  return props.src?.length ? (
    <Image {...props} />
  ) : (
    <KsIcon name={props.fallbackIconName ?? 'info'} />
  )
}
