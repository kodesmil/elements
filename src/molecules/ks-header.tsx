import { KsDropdownMenu } from '@covision/elements/molecules/dashboard/ks-dropdown-menu'
import { KsSheet } from '@covision/elements/molecules/ks-sheet'
import type { ReactNode } from 'react'
import { cn } from '../lib'

export function KsHeader(props: {
  action: ReactNode
  className?: string
}) {
  return (
    <header
      className={cn(
        'sticky top-0 z-30 flex h-14 items-center justify-between gap-4 border-b px-4 py-2 sm:static sm:h-auto sm:border-0 sm:bg-transparent sm:px-6',
        props.className
      )}
    >
      <div className={'flex gap-2'}>
        <KsSheet />
      </div>
      <div className={'flex items-center gap-4'}>
        {props.action}
        <div className={'flex-1'}>
          <KsDropdownMenu />
        </div>
      </div>
    </header>
  )
}
