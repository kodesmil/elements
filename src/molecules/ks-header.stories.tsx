import type { Meta, StoryObj } from '@storybook/react'
import { KsHeader } from './ks-header'

const meta = {
  component: KsHeader,
} satisfies Meta<typeof KsHeader>

export default meta

type Story = StoryObj<typeof meta>

export const Default: Story = {
  args: {
    action: <div>Action</div>,
  },
}
