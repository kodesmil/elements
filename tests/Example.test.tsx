/*
import { render } from '@testing-library/react'
import { expect, it } from 'vitest'
import { KsExample } from '../src/molecules/ks-example'

it('renders correctly', () => {
  const { container } = render(<KsExample text="Clicked this many times" />)
  expect(container).toMatchSnapshot()
})
*/
