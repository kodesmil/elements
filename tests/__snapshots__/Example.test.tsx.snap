// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<div>
  <button
    id="example-button"
    type="button"
  >
    Clicked this many times 0
  </button>
</div>
`;
